overlord/snapstate/snapmgr.go

```diff
@@ -1029,6 +1140,23 @@ func (m *SnapManager) startSnapServices(t *state.Task, _ *tomb.Tomb) error {
 	return err
 }
 
+func scheduleNextRefresh(st *state.State) time.Time {
+	randomness := rand.Int63n(int64(refreshRandomness))
+	nextRefreshTime := time.Now().Add(refreshInterval).Add(time.Duration(randomness))
```

Comment:
This is indeed a bit different from the previous systemd timer. Happy to tweak the values for the store. Or implementing the old behaviour (fixed times plus random time after that fixed time).